packages/lib/storage/service.ts

```diff
@@ -79,6 +119,53 @@ export const getFileFromLocalStorage = async (filePath: string): Promise<TGetFil
   }
 };
 
+// a single service for generating a signed url based on user's environment variables
+export const getUploadSignedUrl = async (
+  fileName: string,
+  environmentId: string,
+  fileType: string,
+  accessType: TAccessType,
+  plan: "free" | "pro" = "free"
+): Promise<TGetSignedUrlResponse> => {
+  // handle the local storage case first
+  if (!IS_S3_CONFIGURED) {
+    try {
+      const { signature, timestamp, uuid } = generateLocalSignedUrl(fileName, environmentId, fileType);
+
+      return {
+        signedUrl: LOCAL_UPLOAD_URL[accessType],
+        signingData: {
+          signature,
+          timestamp,
+          uuid,
+        },
+        fileUrl: new URL(`${WEBAPP_URL}/storage/${environmentId}/${accessType}/${fileName}`).href,
+      };
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  try {
+    const { presignedFields, signedUrl } = await getSignedUrlForS3Upload(
+      fileName,
+      fileType,
+      accessType,
+      environmentId,
+      accessType === "public",
+      plan
+    );
+
+    return {
+      signedUrl,
+      presignedFields,
+      fileUrl: new URL(`${WEBAPP_URL}/storage/${environmentId}/${accessType}/${fileName}`).href,
+    };
+  } catch (err) {
+    throw err;
+  }
+};
+
 export const getSignedUrlForS3Upload = async (
```

Comment:
To stay within one naming schema I think `getUploadSignedUrlS3` would be a better name :-)